src/metabase/db/spec.clj

```diff
@@ -38,3 +38,15 @@
           :subname (str "//" host ":" port "/" db)
           :delimiters "`"}
          (dissoc opts :host :port :db)))
+
+(defn redshift
+  "Create a database specification for a redshift database. Opts should include
+  keys for :db, :user, and :password. You can also optionally set host and
+  port."
+  [{:keys [host port db]
+    :as opts}]
+  (merge {:classname "com.amazon.redshift.jdbc.Driver" ; must be in classpath
+          :subprotocol "redshift"
+          :subname (str "//" host ":" port "/" db "?OpenSourceSubProtocolOverride=false")
+          :ssl true}
+         (dissoc opts :host :port :db)))
```

Comment:
@senior I forgot to mention this during the review but the only reason we have this "spec" namespace is because the code here is used for application DB connections as well. See my note at the top of the namespace:

```clojure
(ns metabase.db.spec
  "Functions for creating JDBC DB specs for a given engine.
   Only databases that are supported as application DBs should have functions in this namespace;
   otherwise, similar functions are only needed by drivers, and belong in those namespaces.")
```

The "spec" stuff is basically just a copy of the [Korma functions](https://github.com/korma/Korma/blob/master/src/korma/db.clj#L190) that did the same thing that I pulled over when we switched from Korma to HoneySQL. (Korma does substantially more than HoneySQL out-of-the-box, so during the transition I had to manually replicate the missing pieces of functionality, such as the C3P0 connection pools. That's also how [Toucan](https://github.com/metabase/toucan) got it's start.)

So we would probably be best served by moving this `redshift` function into `driver/redshift` and renaming it something like `db-details->spec`. See the other JDBC-based drivers for examples

Reply:
Actually explaining this took longer than just fixing it so I went ahead and opened #6204 to move the function